src/server/journal/streamer.cc

```diff
@@ -83,8 +83,11 @@ void RestoreStreamer::Start(io::Sink* dest) {
       }
     } while (cursor);
 
-    WriteCommand(make_pair(
-        "DFLYMIGRATE", ArgSlice{"FULL-SYNC-CUT", absl::StrCat(sync_id_), absl::StrCat(flow_id_)}));
+    VLOG(2) << "FULL-SYNC-CUT for " << sync_id_ << " : " << db_slice_->shard_id();
+    WriteCommand(make_pair("DFLYMIGRATE", ArgSlice{"FULL-SYNC-CUT", absl::StrCat(sync_id_),
+                                                   absl::StrCat(db_slice_->shard_id())}));
+    NotifyWritten(true);
+    is_stable_sync_ = true;
```

Comment:
If the snapshot fiber is created in OutgoingMigration::Flow and it would call a function that does this traverse loop than you would not need the is_stable_sync_ here and OutgoingMigration::Flow could update the state once the functions that wraps this loop finishes.
Anyway if you dont go with this approach I think the is_stable_sync_ name can be canged to snapshot_finished_ as this level should not be aware of what is stable/full sync